### PR TITLE
*: support explain aggregation query.

### PR DESCRIPTION
**Pull request by hanfei1991**
The explain result becomes more precise.

@coocood @shenli @zimulala @tiancaiamao @XuHuaiyu PTAL

### CODE REVIEW

File: executor/explain_test.go [hunk 1]
```diff
@@ -228,6 +184,55 @@ func (s *testSuite) TestExplain(c *C) {
...
+                "[b.a]"
+            ],
+            "child": {
+                "type": "TableScan",
+                "db": "test",
+                "table": "t",
+                "desc": false,
+                "keep order": false,
+                "access condition": null,
+                "limit": 0
```
Comment (coocood) [inline executor/explain_test.go]: This plan should show that aggregation has pushed down.

File: plan/planbuilder.go [hunk 2]
```diff
@@ -452,11 +452,17 @@ func (b *planBuilder) buildExplain(explain *ast.ExplainStmt) Plan {
 			b.err = errors.Trace(err)
 			return nil
 		}
+		solver := &aggPushDownSolver{
+			ctx:   b.ctx,
+			alloc: b.allocator,
+		}
+		solver.aggPushDown(logic)
```
Comment (coocood) [inline plan/planbuilder.go]: Can we call Optimize in this fucntion, so we don't need to maintain two copy of similar code?

**Comment by coocood** [comment]
LGTM

**Comment by XuHuaiyu** [comment]
LGTM

**Pull request merged**